src/wazuh_modules/syscollector/sysInfoMac.cpp

```diff
@@ -272,3 +282,65 @@ nlohmann::json SysInfo::getOsInfo() const
     }
     return ret;
 }
+
+static void getProcessesSocketFD(std::map<ProcessInfo, std::vector<std::shared_ptr<socket_fdinfo>>>& processSocket)
+{
+    int32_t maxProcess { 0 };
+    auto maxProcessLen { sizeof(maxProcess) };
+    sysctlbyname("kern.maxproc", &maxProcess, &maxProcessLen, nullptr, 0);
+
+    auto pids { std::make_unique<pid_t[]>(maxProcess) };
+    const auto processesCount { proc_listallpids(pids.get(), maxProcess) };
+
+    for (auto i = 0 ; i < processesCount ; ++i)
+    {
+        const auto pid { pids.get()[i] };
```

Comment:
not sure if you need the get(), I think you can use directly the index operator.